From 300ca4605df56636f05522bcd34308226b72c00e Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Fri, 6 Feb 2026 10:35:35 -0800
Subject: [PATCH] dummyaudio: Avoid relying on pthreads for Emscripten.

There is no support for SDL_SYS_CreateThread for Emscripten, so this check is completely broken.
---
 src/audio/dummy/SDL_dummyaudio.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/audio/dummy/SDL_dummyaudio.c b/src/audio/dummy/SDL_dummyaudio.c
index b325acb7f..8285153db 100644
--- a/src/audio/dummy/SDL_dummyaudio.c
+++ b/src/audio/dummy/SDL_dummyaudio.c
@@ -25,7 +25,7 @@
 #include "../SDL_sysaudio.h"
 #include "SDL_dummyaudio.h"
 
-#if defined(SDL_PLATFORM_EMSCRIPTEN) && !defined(__EMSCRIPTEN_PTHREADS__)
+#if defined(SDL_PLATFORM_EMSCRIPTEN)
 #include <emscripten/emscripten.h>
 #endif
 
@@ -60,7 +60,7 @@ static bool DUMMYAUDIO_OpenDevice(SDL_AudioDevice *device)
     }
 
     // on Emscripten without threads, we just fire a repeating timer to consume audio.
-    #if defined(SDL_PLATFORM_EMSCRIPTEN) && !defined(__EMSCRIPTEN_PTHREADS__)
+    #if defined(SDL_PLATFORM_EMSCRIPTEN)
     MAIN_THREAD_EM_ASM({
         var a = Module['SDL3'].dummy_audio;
         if (a.timers[$0] !== undefined) { clearInterval(a.timers[$0]); }
@@ -75,7 +75,7 @@ static void DUMMYAUDIO_CloseDevice(SDL_AudioDevice *device)
 {
     if (device->hidden) {
         // on Emscripten without threads, we just fire a repeating timer to consume audio.
-        #if defined(SDL_PLATFORM_EMSCRIPTEN) && !defined(__EMSCRIPTEN_PTHREADS__)
+        #if defined(SDL_PLATFORM_EMSCRIPTEN)
         MAIN_THREAD_EM_ASM({
             var a = Module['SDL3'].dummy_audio;
             if (a.timers[$0] !== undefined) { clearInterval(a.timers[$0]); }
@@ -114,7 +114,7 @@ static bool DUMMYAUDIO_Init(SDL_AudioDriverImpl *impl)
     impl->HasRecordingSupport = true;
 
     // on Emscripten without threads, we just fire a repeating timer to consume audio.
-    #if defined(SDL_PLATFORM_EMSCRIPTEN) && !defined(__EMSCRIPTEN_PTHREADS__)
+    #if defined(SDL_PLATFORM_EMSCRIPTEN)
     MAIN_THREAD_EM_ASM({
         Module['SDL3'].dummy_audio = {};
         Module['SDL3'].dummy_audio.timers = [];
-- 
2.53.0

